README.md

```diff
@@ -151,16 +151,20 @@ The `auth` option can be
    - GitHub App bearer tokens
    - GitHub App installation tokens
 
-2. As object with the properties `username`, `password`, `on2fa`.
+2. As object with the properties `username`, `password`, `on2fa`, `note`, `scopes`.
```

Comment:
I think instead of adding `note` and `scopes`, we should add `token` option which has the same properties as the [Create a new Authorization](https://developer.github.com/v3/oauth_authorizations/#parameters), but in camelCase:

```js
new Octokit({
  auth: {
    username: 'octocat',
    password: 'secret',
    token: {
      note: 'my-token',
      scopes: ['repo'],
      // defaults to documentation URL explaining why tokens are created with Basic auth in Octokit,
      noteUrl: '',
      // Optional: Allow to create a token for an OAuth app
      clientId: 123,
      clientSecret: 'secret',
      fingerprint: 'fingerprint'
    },
    async on2fa () {
      // example: ask the user
      return prompt('Two-factor authentication Code:')
    }
  }
})
```

If `token` is not set, we will start logging a warning with a URL to documentation which explains why, same as the `token.noteUrl` default